### PR TITLE
[Journal] Support Add Journal

### LINKED ISSUE #33: [Journal] Support Add Journal
**Goal:** Add the ability to add journal(s) or journal entries into the journal page:

**Acceptance Criteria:**
- Some sort of button or way to 'click' to add a journal
- The new journal gets created and is displayed on the journal page
- The journal has the core functions of having a title, documentation, and reflection area
- There should be a way to "access" or get into that particular journal
- etc.

### PR DESCRIPTION
**Pull request by kennethkietvuong**
Refer to the GitHub issues linked to this PR for a little bit more detail!

- Created template HTML to host journal page
- Created temporary CSS file for HTML
- Created JS file that contains the main functionality of adding journals onto the journal page

[add-journal-function-UI.webm](https://github.com/cse110-sp24-group30/cse110-sp24-group30/assets/103289124/e734ad47-345a-4c69-903f-9e69c1ee64b1)

### CODE REVIEW

**Review by shjucsd — COMMENTED** [review]
Adding feature works well, if we edit it then the local storage does not update but I think that task is for the people working on edit journal.  (I am not sure if we want to have a pop up when we click on the add journal button and save that in the local storage and then the people for edit journal can work on updating that.)

File: journal/journal-app.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    const journalTemplate = {
+        id: Math.floor(Math.random() * 2000000),
+        title: "",
+        documentation: "",
+        reflection: ""
+    };
+    
+    const modalRef = document.getElementById('modal');
+
+    if (existing) {
```
Comment (adironene) [inline journal/journal-app.js:38]: could you please add a comment about the logic behind the `existing` flag

Comment (adironene) [reply journal/journal-app.js:38]: I understand the logic that if the journal alr exists, then we load the list of journals. But why should the existing flag ever be true to pass in the function?

File: journal/journal-app.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                });
+
+                journalContainer.append(journalElementTest);
+        });
+    }
+    else {
+        journalID = createJournalElement(journalTemplate.id, journalTemplate.title, 
+            journalTemplate.documentation, journalTemplate.reflection, modalRef);
+            
+            const journalElementTest = document.createElement('div');
```
Comment (adironene) [inline journal/journal-app.js:64]: the `else` block has a bit of the repeating code as the block above. Consider simplifying. I know you mentioned sanjith has a more polished version of your code so feel free to ignore if he addressed this

File: journal/journal-page.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <link rel="stylesheet" href="./journal-style.css">
+    <script src="./journal-app.js"></script> <!--Should we use defer keyword?-->
```
Comment (adironene) [inline journal/journal-page.html:8]: please raise this up for discussion to the team so others can help contribute to decision

File: journal/journal-style.css [hunk 4]
```diff
@@ -0,0 +1,108 @@
+body {
```
Comment (adironene) [inline journal/journal-style.css:1]: please attach a video to demonstrate UI

File: journal/journal-app.js (hunk 1 above)
Comment (AliMohammadiasl) [reply journal/journal-app.js:38]: Same feedback, little confused about `existing`.

Comment (kennethkietvuong) [reply journal/journal-app.js:38]: The whole gist of the `existing` flag is to check whether or not there are journals from `localStorage` and then display those onto the journal page. The flag is only `true` when we initially load or attempt to reload the journal page.

This was my approach to utilizing `localStorage` prior to learning to use a database like SQLite. If I didn't include the sections of the `existing` flag, then if we reload the page or come back to the page, you won't see the previously created journals displayed at all (however it is located in `localStorage`).

File: journal/journal-app.js (hunk 2 above)
Comment (kennethkietvuong) [reply journal/journal-app.js:64]: I'm aware that I essentially have copy-and-paste code for the `existing` code section. Like you said, Sanjith and James forked from my branch and extended/optimized it by creating a new function for the copy-and-paste code called `createJournalWidget()`.

File: journal/journal-style.css (hunk 4 above)
Comment (kennethkietvuong) [reply journal/journal-style.css:1]: Video has been attached.

**Review by adironene — APPROVED** [review]

**Pull request merged**